contrib/install.sh

```diff
@@ -0,0 +1,1365 @@
+        ;;
+    -v | --assume-valid)
+        check_interactive_mode
+        assume_valid="$2"
+        shift 2
+        ;;
+    -u | --assume-utreexo)
+        check_interactive_mode
+        assume_utreexo=true
+        shift
+        ;;
+    -s | --ssl)
+        check_interactive_mode
+        enable_ssl=true
+        shift
+        ;;
+    -f | --filters)
+        check_interactive_mode
+        enable_cfilters=true
+        filters_start_height="$2"
+        shift 2
+        ;;
+    -t | --tag)
+        check_interactive_mode
+        defaultTag="$2"
+        tarSrc="https://github.com/$defaultRepo/Floresta/archive/refs/tags/$defaultTag.tar.gz"
+        bdlDir="/tmp/Floresta-$defaultTag"
+        tarDest="$bdlDir.tar.gz"
+        shift 2
+        ;;
+    -p | --proxy)
+        check_interactive_mode
+        proxy="$2"
+        shift 2
+        ;;
+    -c | --connect)
+        check_interactive_mode
+        connect="$2"
+        shift 2
+        ;;
+    -z | --zmq-address)
+        check_interactive_mode
+        zeromq="$2"
+        shift 2
+        ;;
+    -U | --uninstall)
+        check_interactive_mode
+        uninstall_mode=true
+        shift
+        ;;
+    -N | --non-interactive)
+        interactive_mode=false
+        shift
+        ;;
+    -h | --help)
+        show_usage
+        exit 0
+        ;;
+    --)
+        shift
+        break
+        ;;
+    *)
+        echo "Invalid option: $1" >&2
+        show_usage
+        exit 1
+        ;;
+    esac
+done
+
+# func: validate_network
+#
+# This function validates the provided network. If an invalid network is provided, this function
+# will show an error and abort the execution with exit 1
+#
+# Valid networks are:
+#   - bitcoin
+#   - signet
+#   - testnet
+#   - regtest
+validate_network() {
+    valid_networks=("bitcoin" "signet" "testnet" "regtest")
+    if [[ ! " ${valid_networks[@]} " =~ " ${network} " ]]; then
+        echo "❌ Invalid network '$network'. Valid options are: ${valid_networks[*]}"
+        exit 1
+    fi
+}
+
+# func: apt_install
+#
+# Check for missing packages (currently, only debian-like)
+# TODO: verify for fedora, arch and other systems
+apt_install() {
+    local missing=()
+
+    for package in "$@"; do
+        if ! dpkg -l | grep -q "^ii.*$package"; then
+            missing+=("$package")
+        fi
+    done
+
+    # Install curl first
+    sudo apt-get install -y -q curl
+
+    # Install missing packages if any
+    if [ ${#missing[@]} -gt 0 ]; then
+        echo "🐧 The following packages are missing: ${missing[*]}"
+        echo "🐧 Updating package lists..."
+        sudo apt update
+        echo "🐧 Installing missing packages..."
+        sudo apt-get install -y -q "${missing[@]}"
+    else
+        echo "🐧 All packages installed: skip"
+    fi
+}
+
+# func: install_rustup
+#
+# Install rustup with the recommended procedure (https://sh.rustup.rs)
+install_rustup() {
+    echo "🦀 Checking for rustup."
+    haveRust=$(which rustup)
```

Comment:
If we have installed `cargo` through `apt`, I believe we won't have `rustup`. But we should use any `cargo` available in the system. So I think this should be:

```suggestion
    haveRust=$(which cargo)
```
instead